module/VuFind/src/VuFind/Search/ProQuestFSG/Results.php

```diff
@@ -0,0 +1,175 @@
+
+use function strlen;
+
+/**
+ * ProQuest Federated Search Gateway Search Parameters
+ *
+ * @category VuFind
+ * @package  Search_ProQuestFSG
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Page
+ */
+class Results extends \VuFind\Search\Base\Results
+{
+    /**
+     * Search backend identifier.
+     *
+     * @var string
+     */
+    protected $backendId = 'ProQuestFSG';
+
+    /**
+     * Facets returned in search response.
+     *
+     * @var array
+     */
+    protected $responseFacets = null;
+
+    /**
+     * Simplified version of result facets.
+     *
+     * @var array
+     */
+    protected $simplifiedResponseFacets = [];
+
+    /**
+     * Support method for performAndProcessSearch -- perform a search based on the
+     * parameters passed to the object.
+     *
+     * @return void
+     */
+    protected function performSearch()
+    {
+        $query  = $this->getParams()->getQuery();
+        $allTerms = $query->getAllTerms();
+        if (!strlen($allTerms)) {
+            $this->storeErrorResponse('empty_search_disallowed');
+            return;
+        }
+        $limit  = $this->getParams()->getLimit();
+        $offset = $this->getStartRecord();
+        $params = $this->getParams()->getBackendParameters();
+        $command = new SearchCommand(
+            $this->backendId,
+            $query,
+            $offset,
+            $limit,
+            $params
+        );
+        $collection = $this->getSearchService()
+            ->invoke($command)->getResult();
+
+        $this->resultTotal = $collection->getTotal();
+        $this->results = $collection->getRecords();
+
+        // ProQuest does not return facets unless the startRecord is 1
+        if ($offset === 1) {
+            $this->responseFacets = $collection->getFacets();
+            $this->simplifiedResponseFacets = $this->simplifyFacets($this->responseFacets);
+        }
+    }
+
+    /**
+     * Store an empty response with an error message instead of performing a search.
+     *
+     * @param string|array $error Error message(s) to display to user.
+     *
+     * @return void
+     */
+    protected function storeErrorResponse(string|array $error): void
+    {
+        $this->resultTotal = 0;
+        $this->results = [];
+        $this->simplifiedResponseFacets = [];
+        $this->responseFacets = [];
+        $this->errors = (array)$error;
+    }
+
+    /**
+     * Returns the stored list of facets for the last search
+     *
+     * @param array $filter Array of field => on-screen description listing
+     * all of the desired facet fields; set to null to get all configured values.
+     *
+     * @return array        Facets data arrays
+     */
+    public function getFacetList($filter = null)
+    {
+        if (!empty($this->getParams()->getFacetConfig()) && empty($this->simplifiedResponseFacets)) {
```

Comment:
I think this might be the most technically correct solution:

```suggestion
        $activeFacets = $filter ?? $this->getParams()->getFacetConfig();
        if (!empty($activeFacets) && empty($this->simplifiedResponseFacets)) {
```

Reply:
Looks good!